### ISSUE TITLE
Support Multiple Query Include Values

### ISSUE THREAD

**Issue by nowlena** [matthewvolk/bigrequest#84]
Very much liking the package! I think we'll move to using it given it's complete generated type safety approach that should always align with the actual BigCommerce APIs. Well done!

As for this issue... It'd be great to allow multiple "include" values to be used:

```ts
  const product = await bc.v3.GET('/catalog/products/{product_id}', {
    params: {
      path: { product_id: 1 },
      header: { Accept: 'application/json' },
      query: { include: 'variants,images,primary_image' }, // this currently throws an error
    },
  });
```

If you're able to target the query "include" values and make them their own type I believe you could do something like below:

```ts
// generate stand alone types
type GetProductByIdQueryIncludeValue = "variants" | "images" | "custom_fields" | "bulk_pricing_rules" | "primary_image" | "modifiers" | "options" | "videos";
type GetProductByIdQueryInclude = `${GetProductByIdQueryIncludeValue},${GetProductByIdQueryIncludeValue}` | GetProductByIdQueryIncludeValue;

// rest operations
export interface operations {
  getProductById: {
    parameters: {
      query?: {
        include?: GetProductByIdQueryInclude; // use our generated stand-alone type
      };
    };
  };
};
```

I'm sure there's a generic that can be written to improve things but I think you get the point.

Thanks!

**Comment by matthewvolk** [comment]
Thanks so much for the feedback, Aaron! Very happy to hear that the client's been of use to you 🙂 

Apologies for the delay in getting back to you, just got back from a couple of weeks on travel for work. I like your suggestion a lot, and will play around with some generics/the generate script to see if I can automate this! Hope to get back to you soon 👍

**Comment by matthewvolk** [comment]
Hi @nowlena!

I opened a pull request in the bigcommerce/api-specs repository that should fix the issue you raised:

https://github.com/bigcommerce/api-specs/pull/1527

If the PR is accepted, the `include` query parameter will be updated to accept a value of `string[]`, such as:

```ts
const product = await bc.v3.GET("/catalog/products/{product_id}", {
  params: {
    path: { product_id: 77 },
    header: { Accept: "application/json" },
    query: { include: ["options", "images", "primary_image", "variants"] },
  },
});
```

#### Before (error):
![br_84_before](https://github.com/matthewvolk/bigrequest/assets/28374851/9b369edb-1dfa-4a87-9765-4a5d2addc9f3)

#### After (no error):
![br-84-after](https://github.com/matthewvolk/bigrequest/assets/28374851/9f609f7a-e714-41b9-99cb-c8c3992f8c4c)

I wanted to first try to fix the root cause of this issue at its source, hoping that the fix may help other developers using the api-specs repo for their own use cases; but, in the event that my PR can't be merged due to some unintended side effect, I'll re-explore a workaround implemented directly in this library like the one you suggested 🙂

**Comment by matthewvolk** [comment]
Hi again, @nowlena! 👋 

After a [good bit of hard work from the BigCommerce DevDocs team](https://github.com/bigcommerce/docs/pull/221/commits), the team has gone through and fixed this problem for all query parameters in bigcommerce/docs#221.

To incorporate this work, I just [released `bigrequest@0.0.32` to NPM](https://www.npmjs.com/package/bigrequest), and after updating locally, I'm able to send multiple query parameters as a typesafe array:

```ts
bc.v3
  .GET("/catalog/products/{product_id}", {
    params: {
      path: { product_id: 77 },
      header: { Accept: "application/json" },
      query: { include: ["variants", "images", "primary_image"] },
    },
  })
  .then((res) => {
    console.log(res.data);
  });
```

If you have a chance, feel free to test this on your end and let me know if you run into any problems! I'll mark this issue as completed in a couple weeks if I don't hear back.

All the best 🍻

**Comment by nowlena** [comment]
Works perfect!

Thank you and the other BC members who helped out, what a great DX!
